File: src/google_work_agent/api/schemas/diagnostics/create_diagnostic_bundle.py
"""Canonical sanitized diagnostic-bundle wire contracts."""

from typing import Literal

from google_work_agent.api.schemas.model import ApiModel


class CreateDiagnosticBundleRequestV1(ApiModel):
    schema_version: Literal[1]
    command_id: str
    scope: Literal["LAST_24H", "RUN"]
    run_id: str | None = None


class DiagnosticBundleMetadataResponseV1(ApiModel):
    schema_version: Literal[1]
    bundle_ref: str
    scope: Literal["LAST_24H", "RUN"]
    created_at_ms: int
    size_bytes: int


__all__ = ["CreateDiagnosticBundleRequestV1", "DiagnosticBundleMetadataResponseV1"]

File: frontend/src/features/diagnostics/api/create_diagnostic_bundle.ts
import { requestJson } from "../../../api/client";

export type DiagnosticBundleMetadata = { schema_version: 1; bundle_ref: string; scope: "LAST_24H" | "RUN"; created_at_ms: number; size_bytes: number };

export function createDiagnosticBundle(commandId: string, scope: "LAST_24H" | "RUN", runId: string | null): Promise<DiagnosticBundleMetadata> {
  return requestJson("/api/v1/diagnostics/bundles", {
    method: "POST",
    body: { schema_version: 1, command_id: commandId, scope, run_id: scope === "RUN" ? runId : null },
  });
}
